src/js/order_details.ts

```diff
@@ -138,7 +139,9 @@ function extractDetailFromDoc(
       context,
     );
 
-    if ( !x ) {
+    // a gift card purchase does not have a name on the order details URL
```

Comment:
Is that the only reason we could have got here? I am super happy about the log message, but this comment could be a misleading.

Reply:
is there another way? If so, let's add it to the comment!

Reply:
We don't know, and cannot know because it's under Amazon's control - it's possible there is already another way and you and I have just not seen or noticed an order that exhibits that behaviour.